src/board.rs

```diff
@@ -70,6 +70,46 @@ pub fn get_piece_from_char(ch: char) -> Piece {
     }
 }
 
+#[derive(Debug)]
```

Comment:
I think the board file is already growing really big and I need to refactor giant chunks of it out into their own modules. I think we should try to refrain from adding loads of new types to this file. Do you think you can create a new module for board errors?